src/parser/grammar.y

```diff
@@ -36,27 +36,31 @@
 query ::= expr(A). { ctx->root = A; }
 
 expr(A) ::= matchClause(B) whereClause(C) createClause(D) returnClause(E) orderClause(F) limitClause(G). {
-	A = New_AST_Query(B, C, D, NULL, NULL, E, F, G);
+	A = New_AST_Query(B, C, D, NULL, NULL, NULL, E, F, G);
 }
 
 expr(A) ::= matchClause(B) whereClause(C) createClause(D). {
-	A = New_AST_Query(B, C, D, NULL, NULL, NULL, NULL, NULL);
+	A = New_AST_Query(B, C, D, NULL, NULL, NULL, NULL, NULL, NULL);
 }
 
 expr(A) ::= matchClause(B) whereClause(C) deleteClause(D). {
-	A = New_AST_Query(B, C, NULL, NULL, D, NULL, NULL, NULL);
+	A = New_AST_Query(B, C, NULL, NULL, NULL, D, NULL, NULL, NULL);
 }
 
 expr(A) ::= matchClause(B) whereClause(C) setClause(D). {
-	A = New_AST_Query(B, C, NULL, D, NULL, NULL, NULL, NULL);
+	A = New_AST_Query(B, C, NULL, NULL, D, NULL, NULL, NULL, NULL);
 }
 
 expr(A) ::= matchClause(B) whereClause(C) setClause(D) returnClause(E) orderClause(F) limitClause(G). {
-	A = New_AST_Query(B, C, NULL, D, NULL, E, F, G);
+	A = New_AST_Query(B, C, NULL, NULL, D, NULL, E, F, G);
 }
 
 expr(A) ::= createClause(B). {
-	A = New_AST_Query(NULL, NULL, B, NULL, NULL, NULL, NULL, NULL);
+	A = New_AST_Query(NULL, NULL, B, NULL, NULL, NULL, NULL, NULL, NULL);
+}
+
+expr(A) ::= mergeClause(B). {
```

Comment:
I had thought that merge syntax expected a RETURN clause, though I may have misunderstood.

Reply:
In the case of MERGE, RETURN is not mandatory, and in our simplified version, where you can't create more than one pattern RETURN is not that important, at least from might perspective, although it shouldn't be too difficult to have a RETURN clause in addition to MERGE.